src/pcap/cmd/pcap-bosh-cli/main.go

```diff
@@ -38,9 +41,8 @@ type options struct {
 	PcapAPIURL         string   `short:"u" long:"pcap-api-url" description:"The URL of the PCAP API, e.g. pcap.cf.$LANDSCAPE_DOMAIN" env:"PCAP_API" required:"true"`
 	Filter             string   `short:"f" long:"filter" description:"Allows to provide a filter expression in pcap filter format." required:"false"`
 	Interface          string   `short:"i" long:"interface" description:"Specifies the network interface to listen on." default:"eth0" required:"false"`
-	Type               string   `short:"t" long:"type" description:"Specifies the type of process to capture for the app." default:"web" required:"false"`
 	BoshConfigFilename string   `short:"c" long:"bosh-config" description:"Path to the BOSH config file, used for the UAA Token" default:"${HOME}/.bosh/config" required:"false"`
-	BoshEnvironment    string   `short:"e" long:"bosh-environment" description:"The BOSH environment to use for retrieving the BOSH UAA token from the BOSH config file" default:"bosh" required:"false"`
+	BoshEnvironment    string   `short:"e" long:"bosh-environment" description:"The BOSH environment to use for retrieving the BOSH UAA token from the BOSH config file" env:"BOSH_ENVIRONMENT" required:"false"`
```

Comment:
Are `env:` and `default:` mutually exclusive? If not, it would be better to support both.

I rarely set `BOSH_ENVIRONMENT` and instead call `bosh -e the_env_that_i_want -d ...`. Not having to worry about it and defaulting to `bosh`, even when the env variable is not set would still be nice.

If it's mutually exclusive, I agree that the env variable has higher precedence.

Reply:
I've had both and it wouldn't read my env var. So I think they are mutually exclusive. Anyway, we can't make any default assumptions on the name of a BOSH env. The BOSH cli also does not do this, it tries to read this env var or expects an explicit parameter, otherwise throw an error. I've tried to align on that behavior here.